pulpcore/pulpcore/app/viewsets/content.py

```diff
@@ -79,3 +81,24 @@ class ArtifactViewSet(CreateDestroyReadNamedModelViewSet):
     queryset = Artifact.objects.all()
     serializer_class = ArtifactSerializer
     filter_class = ArtifactFilter
+
+    def destroy(self, request, pk):
+        """
+        Remove :class:`~pulpcore.app.models.Artifact` only if it doesn't belong to any
+        `~pulpcore.app.models.Content`
+
+        Args:
+            request (:class:`rest_framework.request.Request`): request containing JSON payload
+                with information about the artifact being deleted.
+            pk (:class:`uuid.UUID`): Unique id of the artifact being deleted.
+
+        Returns:
+            :class:`rest_framework.response.Response` with a 204 status code in case of
+                successful deletion, 409 status code if deletion is not allowed.
+        """
+        try:
+            return super(ArtifactViewSet, self).destroy(request, pk)
+        except models.ProtectedError:
+            msg = _('The Artifact cannot be deleted because it is a part of the Content.')
+            data = {'detail': msg}
+            return Response(data, status=status.HTTP_409_CONFLICT)
```

Comment:
Suggest:

s/detail/reason/

Reply:
I called it `detail` becasue DRF uses this terminology in many cases: https://github.com/encode/django-rest-framework/blob/15024f3f07b9311887ad6f5afb366f6c032c0486/rest_framework/views.py#L83-L97

Reply:
Ah, okay.